enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/consensus/shipping/RaftLogShipper.java

```diff
@@ -350,22 +330,21 @@ private boolean timedOut()
 
     private void scheduleTimeout( long deltaMillis )
     {
-        // TODO: This cancel/create dance is a bit inefficient... consider something better.
-
         timeoutAbsoluteMillis = clock.millis() + deltaMillis;
 
-        if ( timeout != null )
+        if ( timer == null )
```

Comment:
Is this the same logic? The former timer doesn't get cancelled I think

Reply:
Interesting. Will investigate what is up here.

Reply:
Yes, so this works a bit differently now. What you are looking at here is the idempotent operation of creating the timer (once). There is no need to cancel() the Timer with this new TimerService when you set() or reset() it.